docs/testing-plan.md

```diff
@@ -0,0 +1,130 @@
+
+- Batch Testing
+
+- Automated Regression Testing
+
+- Beta Testing
+This should take place when we are reviewing the PRs and spin up the application locally.
+
+# 5.0 Hardware Requirements
+Currently we only support spinning up development enviroments on Mac books.
+
+# 6.0 Features to Be Tested
+Note as this application is constnatly being iterated on this features to be tested list should always be reviewed and edited based on features added / removed or updated.
+
+These descriptions are at a high level and the best indicator when testing development is to view what the application looks like in productions.
+
+Remember to test all features in both English and Japanese version of the Docs site.
+
+## Accessibility testing
+- Can you search in algolia?
+- Are the tabs in search toggleable?
+- Can you tab to `Skip to Content`?
+
+## Header
+- CircleCi logo that when clicked takes you to docs homepage
+- Search bar (See search bar below more details)
+- Globe icon that opens to langauge selector (See language selector below for more details)
+- Link to circleci.com
+- Button to `Go to application` or `Sign up`
+
+## Search bar
+- When typing inside of search bar then results a results page should appear. The url should now contain a query string of the word used. 3 tabs should appear
+1. Documention - results found in /docs
+2. Orbs - Results found in the developer/orb site
+3. Convenience Images - Results found in developer/orb
+- Pressing cancel should close search bar results 
+- Pressing `X` should clear search results and query string from url
+
+## Side nav
+- Contains a list to all different resource available on the Docs site
+- There should be top level sections that toggle between open and closed when clicked on revealing child lists 
+- The side nav should ensure the parent category is opened and the child side page header is highlighted in the list for the associated page that the user is currently viewing
+- When an item is clicked on the side nav it should navigate the user to the associated page, and highlight the item should be highlighted
+
+## TOC
+- TOC may not appear on all pages
+- TOC contains a list of headers that are on the page and helpful resources
+- When a user clicks on a helpful resource the user should be navigated to that resource
+- Wehn a user clicked on an item in `On this page` then the user should be scrolled to that item on the page and the item should now be highlighted
+
+## Language selector
+- Language selector allows a user to switch between English and Japanese
+
+## Screen size
+- The header will change depending on screen size. For example the circle ci home page may no longer be visible. 
+- The side nav moves from the left hand side to be used inside of a hamburger icon
+-  TOC moves from right side of the screen to being embeded at the top of the document
+
+# 7.0 Features Not to Be Tested
+- circleci.com
+- sign up / signin
```

Comment:
I think we should at least make sure the buttons accurately reflect user status.